- Description: iucv: pathid mismatch for multiple iucv connections
  Symptom:     Lost IUCV-connections if a further Connection Pending request
               arrives.
               Message 'iucv0: IR pathid 1 does not match original pathid 0'
               in dmesg.
  Problem:     Bug in handler scan in case of a 'Connection Pending' interrupt
  Solution:    reset handler variable h to NULL if scanned handler does not match
  Problem-ID:  9759
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/net/iucv.c	21 May 2004 05:21:35 -0000	1.27.2.3
+++ linux-2.5/drivers/s390/net/iucv.c	1 Jul 2004 14:35:37 -0000	1.27.2.4
@@ -1,5 +1,5 @@
 /* 
- * $Id: iucv.c,v 1.27.2.3 2004/05/21 05:21:35 braunu Exp $
+ * $Id: iucv.c,v 1.27.2.4 2004/07/01 14:35:37 braunu Exp $
  *
  * IUCV network driver
  *
@@ -29,7 +29,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  *
- * RELEASE-TAG: IUCV lowlevel driver $Revision: 1.27.2.3 $
+ * RELEASE-TAG: IUCV lowlevel driver $Revision: 1.27.2.4 $
  *
  */
 
@@ -352,7 +352,7 @@
 static void
 iucv_banner(void)
 {
-	char vbuf[] = "$Revision: 1.27.2.3 $";
+	char vbuf[] = "$Revision: 1.27.2.4 $";
 	char *version = vbuf;
 
 	if ((version = strchr(version, ':'))) {
@@ -2368,7 +2368,8 @@
 					iucv_debug(2,
 						   "found a matching handler");
 					break;
-				}
+				} else
+					h = NULL;
 			}
 			spin_unlock_irqrestore (&iucv_lock, flags);
 			if (h) {

